public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -107,32 +109,32 @@ block what-should-we-use-as-token
       Dart doesn't have this limitation;
       every class implicitly defines an interface,
       so interface names are just class names.
-
-  :marked
-    The problem with using `Map` as a token, is not that it is an abstract
-    class, but rather that it is very general purpose (like `String`) and so
-    our application may depend on several maps each used for a
-    different purpose.
+      `Map` is a *valid* token even though it's the name of an abstract class;
+      it's just *unsuitable* as a token because it's too general.
 
 block dart-map-alternative
   :marked
     As an alternative to using a configuration `Map`, we can define
-    a custom class type:
+    a custom configuration class:
 
   +makeExample('dependency-injection/ts/app/app.config.ts','config-alt','app/app-config.ts (alternative config)')(format='.')
 
   :marked
-    One of the key benefits of defining a configuration object class, is
-    strong static checking relative to configuration property names and their types.
-    The Dart [cascade operator][cascade-op] provides a convenient means of initializing
-    the configuration object. Use of the cascaded object initialization idiom
+    Defining a configuration class has a few benefits. One key benefit
+    is strong static checking: we'll be warned early if we misspell a property
+    name or assign it a value of the wrong type.
+    The Dart [cascade notation][cascade] (`..`) provides a convenient means of initializing
+    a configuration object. 
+
+    Use of the cascaded initialization idiom
```

Comment:
On the right track, but this could still be simplified. Maybe:

If we use cascades, the configuration object can't be declared `const` and we can't use
a [value provider](...). The solution is to use a [factory provider](...):